ros2bag/ros2bag/api/__init__.py

```diff
@@ -129,3 +130,65 @@ def add_standard_reader_args(parser: ArgumentParser) -> None:
         '-s', '--storage', default='', choices=reader_choices,
         help='Storage implementation of bag. '
              'By default attempts to detect automatically - use this argument to override.')
+
+
+def add_storage_plugin_extensions(parser: ArgumentParser) -> None:
+    plugin_choices = set(rosbag2_py.get_registered_writers())
+    storage_parser = ArgumentParser(add_help=False)
+    storage_parser.add_argument(
+        '-s', '--storage',
+        default=rosbag2_py.get_default_storage_id(),
+        choices=plugin_choices,
+        help='Storage implementation of bag. '
+            'By default attempts to detect automatically - use this argument to override.')
+    storage_parsed_args, _ = storage_parser.parse_known_args()
+    plugin_id = storage_parsed_args.storage
+
+    if plugin_id not in plugin_choices:
+        raise ValueError(f'No storage plugin found with ID {plugin_id}')
+
+    pkg_name = f'rosbag2_storage_{plugin_id}'
```

Comment:
do we have no way to extract the package name from a registered plugin otherwise?

Reply:
Yes, I've found a way. The only catch is that the python module must be equal to the containing package name (e.g. has to be called `rosbag2_storage_mcap`). I think this is a reasonable requirement, you agree?

Reply:
Second update: i've changed it to use setuptools' entry_points, so no a priori knowledge is required here at all, any package can register an extension on  its side.